transport/awssqs/producer.go

```diff
@@ -0,0 +1,127 @@
+package awssqs
+
+import (
+	"context"
+	"encoding/json"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/sqs"
+	"github.com/aws/aws-sdk-go/service/sqs/sqsiface"
+	"github.com/go-kit/kit/endpoint"
+)
+
+// Producer wraps an SQS client and queue, and provides a method that
+// implements endpoint.Endpoint.
+type Producer struct {
+	sqsClient        sqsiface.SQSAPI
+	queueURL         string
+	responseQueueURL string
+	enc              EncodeRequestFunc
+	dec              DecodeResponseFunc
+	before           []ProducerRequestFunc
+	after            []ProducerResponseFunc
+	timeout          time.Duration
+}
+
+// NewProducer constructs a usable Producer for a single remote method.
+func NewProducer(
+	sqsClient sqsiface.SQSAPI,
+	queueURL string,
+	responseQueueURL string,
```

Comment:
So, the same question remains. Why do you want to store responseQueueURL in the Producer structure? If someone wants to pass this into message attributes, he will do it in encode or in the beforeProducer. This `responseQueueURL` is not used anywhere.

Reply:
@0marq I understand. You have current project which need the queue url inside the process.

It can be done via plugin before or after function, and context value.

For example:
```go
type contextKey int

const (
	ContextKeyQueueURL contextKey = iota
)
// set this as before or after function
func SetProducerQueueUrl(url string) ProducureRequestFunc {
  return func(ctx context.Context, anotherParams) context.Context {
    return context.WithValue(ctx, ContextKeyQueueURL, url)
  }
}
```

then you can get the value via `ctx.Value(contextKey).(<yourtype>)`

Reply:
The value is actually used when calling before and after